include/deal.II/lac/sparsity_tools.h

```diff
@@ -292,6 +292,28 @@ namespace SparsityTools
                               const MPI_Comm &             mpi_comm,
                               const IndexSet &             myrange);
 
+  /**
+   * Gather rows in a dynamic sparsity pattern over MPI.
+   *
+   * @param dsp A dynamic sparsity pattern that has been built locally and which
+   * we need to extend according to the sparsity of rows stored on other MPI
+   * processes.
+   *
+   * @param rows_per_cpu A vector containing the number of of rows per CPU for
+   * determining ownership. This input should be the same on all MPI processes.
```

Comment:
Could you say how one would obtain this argument.

In fact, I'm not so happy with the way this argument is structured. It makes the implicit assumption that every process stores a contiguous range of rows, but we're tried to get away from this model and the way we initialize these objects these days is via `IndexSet` objects that may or may not be contiguous. I'd be much happier if the argument passed here were simply a vector of IndexSets. Assuming that these would correspond to locally owned DoFs, one can get this vector from DoFHandler.

Reply:
> I'd be much happier if the argument passed here were simply a vector of IndexSets

I can take a `const IndexSet & owned_rows` and deduce what I need form there. I will still support only contiguous ranges at this point.

Reply:
That's ok. Someone could then later implement the non-contiguous case, but at least we wouldn't have to change the interface. In the meantime, you can just check `index_set.is_contiguous()`.

Vector vs single index set: That sounds reasonable. But doesn't the current object knows its local index set already?

Reply:
> I can take a `const IndexSet & owned_rows` and deduce what I need form there. I will still support only contiguous ranges at this point. Would you be ok with this solution?

Or a std::vector<IndexSet> as returned by ``locally_owned_dofs_per_processor()``

Reply:
> a std::vector as returned by locally_owned_dofs_per_processor()

i intend to use it outside of DoFHandler context, so I would rather not take an `std::vector<IndexSet>` here.

Reply:
Nevermind, i will update and add an assert.